### PR TITLE
Import Variants in Traits and settings for it

### PR DESCRIPTION
**Pull request by Maniasnikos**
This is my first commit, let me know if something doesn't feel right. -- Ralakos

### CODE REVIEW

File: js/5etools-monsters.js [hunk 1]
```diff
@@ -1120,6 +1120,27 @@ function d20plusMonsters () {
 									character.attribs.create({name: `repeating_npctrait_${newRowId}_desc`, current: text});
 								});
 							}
+							if (data.variant && d20plus.cfg.getOrDefault("import", "importVariants")) {
+								$.each(data.variant, function (i, v) {
```
Comment (TheGiddyLimit) [inline js/5etools-monsters.js]: Prefer JS' `data.variant.forEach((v, i) => { ... })` over jQuery methods

File: js/5etools-monsters.js [hunk 2]
```diff
@@ -1120,6 +1120,27 @@ function d20plusMonsters () {
...
+									let newRowId = d20plus.ut.generateRowId();
```
Comment (TheGiddyLimit) [inline js/5etools-monsters.js]: Prefer `const` over `let`

File: js/5etools-monsters.js [hunk 3]
```diff
@@ -1120,6 +1120,27 @@ function d20plusMonsters () {
...
+									character.attribs.create({
+										name: `repeating_npctrait_${newRowId}_name`,
+										current: d20plus.importer.getCleanText(renderer.render('Variant : '+v.name)),
+									});
+
+									if (d20plus.cfg.getOrDefault("import", "importVariants")) {
+										const offsetIndex = data.spellcasting ? 1 + i : i;
```
Comment (TheGiddyLimit) [inline js/5etools-monsters.js]: `data.spellcasting` is an array which can be >1 item long; why not `const offsetIndex = (data.spellcasting?.length || 0)+ i;`?

File: js/5etools-monsters.js [hunk 4]
```diff
@@ -1120,6 +1120,27 @@ function d20plusMonsters () {
...
+										character.abilities.create({
+											name: `${offsetIndex}: ${v.name}`,
+											istokenaction: true,
+											action: d20plus.actionMacroTrait(offsetIndex),
+										});
+									}
+
+									let text = d20plus.importer.getCleanText(renderer.render({entries: v.entries}, 1));
```
Comment (TheGiddyLimit) [inline js/5etools-monsters.js]: `let` -> `const`

**Review by TheGiddyLimit — CHANGES_REQUESTED** [review]
Looks good, some minor tweaks needed!

File: js/5etools-monsters.js (hunk 1 above)
Comment (Maniasnikos) [reply js/5etools-monsters.js]: Fixed, noted.

File: js/5etools-monsters.js (hunk 2 above)
Comment (Maniasnikos) [reply js/5etools-monsters.js]: Fixed, noted.

File: js/5etools-monsters.js (hunk 3 above)
Comment (Maniasnikos) [reply js/5etools-monsters.js]: Fixed, most code was copy pasted from 1102-1122 , import traits code. should i change those too?

File: js/5etools-monsters.js (hunk 4 above)
Comment (Maniasnikos) [reply js/5etools-monsters.js]: fixed.

**Comment by Maniasnikos** [comment]
Had wrong account in my Github Desktop, Cretatour is my work account, its still me :) Please use Ralakos as my name

**Review by TheGiddyLimit — APPROVED** [review]

**Pull request merged**